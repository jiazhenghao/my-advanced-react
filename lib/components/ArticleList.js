import React from "react";

import ArticleContainer from "./Article";

// const ArticleList = props => {
//   return (
//     <div>
//       {Object.values(props.articles).map(article => (
//         // <Article key={article.id} article={article} store={props.store} />
//         <ArticleContainer key={article.id} article={article} />
//       ))}
//     </div>
//   );
// };

class ArticleList extends React.PureComponent {
  render() {
    return (
      <div>
        {Object.values(this.props.articles).map(article => (
          <ArticleContainer key={article.id} article={article} />
        ))}
      </div>
    );
  }
}

export default ArticleList;
