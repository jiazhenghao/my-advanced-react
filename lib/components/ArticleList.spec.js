import React from "react";
import ArticleList from "./ArticleList";
import { shallow } from "enzyme";

describe("ArtilcList", () => {
  const testProps = {
    articles: {
      a: {
        id: "a",
        date: "Fri Jul 22 2016 00:00:00 GMT+0000 (UTC)",
        title: "Create Apps with No Configuration",
        body: "This month marks a year since we released Relay and we'd like to"
      },
      b: {
        id: "b",
        date: "Fri Jul 22 2016 00:00:00 GMT+0000 (UTC)",
        title: "Create Apps with No Configuration",
        body: "This month marks a year since we released Relay and we'd like to"
      }
    }
  };

  it("renders correctly", () => {
    const wrapper = shallow(<ArticleList {...testProps} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.children().length).toBe(2);
    //expect(wrapper.find("WithStore").length).toBe(2);
    expect(wrapper.find("ArticleContainer").length).toBe(2);
  });
});
