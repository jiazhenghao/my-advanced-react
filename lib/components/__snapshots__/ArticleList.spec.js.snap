// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtilcList renders correctly 1`] = `ShallowWrapper {}`;
